source/common/formatter/substitution_formatter.cc

```diff
@@ -62,18 +62,26 @@ const std::string SubstitutionFormatUtils::DEFAULT_FORMAT =
     "\"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"\n";
 
 FormatterPtr SubstitutionFormatUtils::defaultSubstitutionFormatter() {
-  return FormatterPtr{new FormatterImpl(DEFAULT_FORMAT)};
+  return FormatterPtr{new FormatterImpl(DEFAULT_FORMAT, false)};
 }
 
-const std::string&
+const absl::optional<std::string>
 SubstitutionFormatUtils::protocolToString(const absl::optional<Http::Protocol>& protocol) {
   if (protocol) {
     return Http::Utility::getProtocolString(protocol.value());
   }
-  return UnspecifiedValueString;
+  return absl::nullopt;
 }
 
-const std::string SubstitutionFormatUtils::getHostname() {
+const std::string
+SubstitutionFormatUtils::protocolToStringOrDefault(const absl::optional<Http::Protocol>& protocol) {
```

Comment:
Is this used anywhere? If not, we should remove it.

Reply:
Yeah, in header_formatter and tracer instead of the old protocolToString as a convenience.